File: EDR_NDR/frontend/src/types/alert.ts
import type { AgentRef } from './agent'

export type Severity = 'critical' | 'high' | 'medium' | 'low' | 'info'
export type AlertStatus = 'new' | 'investigating' | 'resolved' | 'false_positive' | 'closed'
export type AlertSource = 'EDR' | 'NDR'

export interface AlertSummary {
  id: string
  title: string
  severity: Severity
  status: AlertStatus
  source: AlertSource
  agent: AgentRef | null
  src_ip: string | null
  dest_ip: string | null
  mitre_technique: string | null
  assignee: string | null
  created_at: string
  updated_at: string
}

export interface AlertDetail extends AlertSummary {
  description: string
  detection_rule_id: string | null
  rule: { id: string; name: string; rule_type: string; severity: Severity } | null
}

export interface AlertStatsSummary {
  by_severity: Record<string, number>
  by_status: Record<string, number>
  total_open: number
}

export interface AlertListParams {
  q?: string
  severity?: Severity[]
  status?: AlertStatus[]
  source?: AlertSource
  agent_id?: string
  sort?: string
  page?: number
  page_size?: number
}

export interface AlertUpdatePayload {
  status?: AlertStatus
  severity?: Severity
  assignee?: string
}

File: EDR_NDR/backend/app/schemas/alert.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict

from app.models.enums import AlertSource, AlertStatus, Severity
from app.schemas.agent import AgentRef
from app.schemas.rule import RuleRef


class AlertSummary(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: str
    title: str
    severity: Severity
    status: AlertStatus
    source: AlertSource
    agent: AgentRef | None
    src_ip: str | None
    dest_ip: str | None
    mitre_technique: str | None
    assignee: str | None
    created_at: datetime
    updated_at: datetime


class AlertDetail(AlertSummary):
    description: str
    rule: RuleRef | None
    detection_rule_id: str | None


class AlertUpdate(BaseModel):
    status: AlertStatus | None = None
    severity: Severity | None = None
    assignee: str | None = None


class AlertStatsSummary(BaseModel):
    by_severity: dict[str, int]
    by_status: dict[str, int]
    total_open: int
